cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinderTest.java

```diff
@@ -0,0 +1,116 @@
+  }
+
+  @Test
+  public void testDetectingSlowBrokerFromPeer() {
+    SlowBrokerFinder slowBrokerFinder = createSlowBrokerFinder();
+    Map<BrokerEntity, ValuesAndExtrapolations> currentMetrics = new HashMap<>(BROKER_ENTITIES.size());
+    Map<BrokerEntity, ValuesAndExtrapolations> history = new HashMap<>(BROKER_ENTITIES.size());
+    currentMetrics.putAll(createCurrentMetrics(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS * 11),
+                          11, BROKER_ENTITIES.get(0)));
+    history.putAll(createHistory(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS * 11),
+                   10, BROKER_ENTITIES.get(0)));
+    for (int i = 1; i < BROKER_ENTITIES.size(); i++) {
+      currentMetrics.putAll(createCurrentMetrics(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS),
+                            11, BROKER_ENTITIES.get(i)));
+      history.putAll(createHistory(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS),
+                     10, BROKER_ENTITIES.get(i)));
+    }
+    Collection<MetricAnomaly<BrokerEntity>> anomalies = slowBrokerFinder.metricAnomalies(history, currentMetrics);
+    assertTrue("There should be exactly a single slow broker", anomalies.size() == 1);
+    MetricAnomaly<BrokerEntity> anomaly = anomalies.iterator().next();
+    assertTrue(anomaly.entities().containsKey(BROKER_ENTITIES.get(0)));
+    assertEquals(ANOMALY_DETECTION_TIME_MS, (long) anomaly.entities().get(BROKER_ENTITIES.get(0)));
+  }
+
+  @Test
+  public void testExcludingSmallTrafficBroker() {
+    SlowBrokerFinder slowBrokerFinder = createSlowBrokerFinder();
+    Map<BrokerEntity, ValuesAndExtrapolations> currentMetrics = new HashMap<>(BROKER_ENTITIES.size());
+    Map<BrokerEntity, ValuesAndExtrapolations> history = new HashMap<>(BROKER_ENTITIES.size());
+    currentMetrics.putAll(
+        createCurrentMetrics(populateMetricValues(SMALL_BYTES_IN_RATE, SMALL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS), 11, BROKER_ENTITIES.get(0)));
+    history.putAll(
+        createHistory(populateMetricValues(SMALL_BYTES_IN_RATE, SMALL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS), 10, BROKER_ENTITIES.get(0)));
+    for (int i = 1; i < BROKER_ENTITIES.size(); i++) {
+      currentMetrics.putAll(createCurrentMetrics(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS),
+                            11, BROKER_ENTITIES.get(i)));
+      history.putAll(
+          createHistory(populateMetricValues(NORMAL_BYTES_IN_RATE, NORMAL_BYTES_IN_RATE, NORMAL_LOG_FLUSH_TIME_MS), 10, BROKER_ENTITIES.get(i)));
+    }
+    Collection<MetricAnomaly<BrokerEntity>> anomalies = slowBrokerFinder.metricAnomalies(history, currentMetrics);
+    assertTrue(anomalies.isEmpty());
+  }
+
+  @Test
+  public void testInsufficientData() {
```

Comment:
(applies to other unit tests) The tested scenario is not clear, can we add comments or JavaDoc to give a high-level overview of what is being tested?